### PR TITLE
EscrowCreate, EscrowFinish and EscrowCancel

### PR DESCRIPTION
**Pull request by florent-uzio**
# Title

This PR aims to add the Flatten and Validate func for `EscrowCreate`, `EscrowCancel` and `EscrowFinish`.
It also adds the tests, the three transaction types have 100% test coverage.

### CODE REVIEW

File: xrpl/transaction/escrow_cancel.go [hunk 1]
```diff
@@ -1,20 +1,70 @@
...
+	if e.OfferSequence != 0 {
+		flattened["OfferSequence"] = e.OfferSequence
+	}
+
+	return flattened
+}
+
+// Validate checks if the EscrowCancel struct is valid.
+func (e *EscrowCancel) Validate() (bool, error) {
+	_, err := e.BaseTx.Validate()
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_cancel.go:56]: validate return bool should not be ignored

File: xrpl/transaction/escrow_cancel_test.go [hunk 2]
```diff
@@ -1 +1,149 @@
...
+				},
+				Owner: "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
+			},
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			valid, _ := tt.escrow.Validate()
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_cancel_test.go:145]: error should be checked

File: xrpl/transaction/escrow_cancel_test.go [hunk 3]
```diff
@@ -1 +1,149 @@
...
+		})
+	}
+}
+
+func TestEscrowCancel_Validate(t *testing.T) {
+	tests := []struct {
+		name     string
+		escrow   *EscrowCancel
+		expected bool
+	}{
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_cancel_test.go:94]: We should add support to check error cases

File: xrpl/transaction/escrow_create.go [hunk 4]
```diff
@@ -1,24 +1,96 @@
...
+	}
+	if e.DestinationTag != 0 {
+		flattened["DestinationTag"] = e.DestinationTag
+	}
+
+	return flattened
+}
+
+// Validates the EscrowCreate transaction and makes sure all the fields are correct.
+func (e *EscrowCreate) Validate() (bool, error) {
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_create.go:78]: Does this method validate the following fields combination:

Screenshot 2024-11-04 at 09 28 17

File: xrpl/transaction/escrow_create_test.go [hunk 5]
```diff
@@ -1 +1,179 @@
...
+			}
+		})
+	}
+}
+
+func TestEscrowCreate_Validate(t *testing.T) {
+	tests := []struct {
+		name     string
+		entry    *EscrowCreate
+		expected bool
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_create_test.go:78]: Add `expectedErr` field

File: xrpl/transaction/escrow_create_test.go [hunk 6]
```diff
@@ -1 +1,179 @@
...
+				CancelAfter: 533257958,
+			},
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			valid, _ := tt.entry.Validate()
+			assert.Equal(t, tt.expected, valid)
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_create_test.go:176]: Check `Validate` error cases

File: xrpl/transaction/escrow_finish_test.go [hunk 7]
```diff
@@ -1 +1,144 @@
...
+			}
+		})
+	}
+}
+
+func TestEscrowFinish_Validate(t *testing.T) {
+	tests := []struct {
+		name     string
+		entry    *EscrowFinish
+		expected bool
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_finish_test.go:88]: Add error cases and `expectedErr` field

File: xrpl/transaction/escrow_finish_test.go [hunk 8]
```diff
@@ -1 +1,144 @@
...
+				Owner: "rf1BiGeXwwQoi8Z2ueFYTEXSwuJYfV2Jpn",
+			},
+			expected: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			valid, _ := tt.entry.Validate()
+			assert.Equal(t, tt.expected, valid)
```
Comment (GuillemGarciaDev) [inline xrpl/transaction/escrow_finish_test.go:141]: Add error validation

**Review by GuillemGarciaDev — CHANGES_REQUESTED** [review]

File: xrpl/transaction/escrow_cancel.go (hunk 1 above)
Comment (florent-uzio) [reply xrpl/transaction/escrow_cancel.go:56]: Updated

File: xrpl/transaction/escrow_cancel_test.go (hunk 3 above)
Comment (florent-uzio) [reply xrpl/transaction/escrow_cancel_test.go:95]: I've added a `wantErr` and replaced `expected` by `wantValid`.

File: xrpl/transaction/escrow_create.go (hunk 4 above)
Comment (florent-uzio) [reply xrpl/transaction/escrow_create.go:78]: Yes, I've added more tests to make sure but all looks good.

**Review by GuillemGarciaDev — APPROVED** [review]

**Pull request merged**